### PR TITLE
Dj app

### PR DESCRIPTION
**Pull request by geremyCohen**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: examples/apps/djapp/5_canary/metal_v2.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      version: v2
+  template:
+    metadata:
+      labels:
+        app: metal
+        version: v2
+    spec:
+      containers:
+        - name: metal
+          image: "672518094988.dkr.ecr.us-west-2.amazonaws.com/hello-world:v1.0"
```
Comment (bcelenza) [inline examples/apps/djapp/5_canary/metal_v2.yaml]: Are the images mentioned in this example publicly available from this repository? Do customers need to build and push their own versions of these?

**Comment by geremyCohen** [comment]
They are publically available

File: examples/apps/djapp/5_canary/metal_v2.yaml (hunk 1 above)
Comment (geremyCohen) [reply examples/apps/djapp/5_canary/metal_v2.yaml]: The images are hosted in a dedicated account.

**Comment by efe-selcuk** [comment]
I don't see how the initial mesh resources are created but I do see the delete script. Would be awesome if we could get these changed to cloudformation resources like colorapp.

File: examples/apps/djapp/5_canary/metal_v2.yaml (hunk 1 above)
Comment (geremyCohen) [reply examples/apps/djapp/5_canary/metal_v2.yaml]: @bcelenza https://github.com/aws/aws-app-mesh-examples/pull/129 adds the README and removes the .gitignore.

**Comment by brentley** [comment]
Cloudformation would imply some opinion on deployment and target...it would be nice to keep this example neutral, and inject the opinion at the workshop or blog level that will reference these apps, like eksworkshop and eksworkshop.

File: examples/apps/djapp/3_add_crds/virtual-node-definition.yaml [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+                      port:
+                        type: integer
+                      protocol:
+                        type: string
+                        enum:
+                          - tcp
+                          - http
+                          - grpc
+                          - http2
+                          - https
```
Comment (bcelenza) [inline examples/apps/djapp/3_add_crds/virtual-node-definition.yaml]: Hmm, these are not all currently valid (https://docs.aws.amazon.com/app-mesh/latest/APIReference/API_PortMapping.html). 

Brings up a general question of how we intend to modify this demo as new features are added?

File: examples/apps/djapp/3_add_crds/virtual-node-definition.yaml [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+            serviceDiscovery:
+              type: object
+              properties:
+                cloudMap:
```
Comment (bcelenza) [inline examples/apps/djapp/3_add_crds/virtual-node-definition.yaml]: Same here -- Cloud Map is not yet supported, so this should not be an option.

File: examples/apps/djapp/3_add_crds/virtual-service-definition.yaml [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+              type: object
+              properties:
+                name:
+                  type: string
+            routes:
+              type: array
+              items:
+                type: object
+                properties:
+                  http:
```
Comment (bcelenza) [inline examples/apps/djapp/3_add_crds/virtual-service-definition.yaml]: Will this demo ever include TCP routes?

**Comment by efe-selcuk** [comment]
> Cloudformation would imply some opinion on deployment and target...it would be nice to keep this example neutral, and inject the opinion at the workshop or blog level that will reference these apps, like eksworkshop and eksworkshop.

To be clear, I only wanted cloudformation for the App Mesh resources, but if that's undesirable too, then that's cool. Creation/deletion with just scripts can be a bit painful given again the eventual consistency.

**Pull request merged**